src/vegur_proxy.erl

```diff
@@ -766,7 +766,7 @@ check(Transport, Sock) ->
         true -> % no check after buffer is full, let the kernel handle it.
             ok;
         false ->
-            case Transport:recv(Sock, 0, 0) of
+            case Transport:recv(Sock, 0, 1) of
```

Comment:
Uh, I think there's a big distinction here on whether the thing is gonna get descheduled or not if you wait that way.

I'd rather have us special-case SSL than also throw TCP under the bus there.